site/app/views/forum/ForumThreadView.php

```diff
@@ -919,81 +878,86 @@ public function createPost($thread_id, $post, $unviewed_posts, $function_date, $
 					$ud_button_title = "Remove post";
 					$ud_button_icon = "fa-trash";
 				}
-				$return .= <<<HTML
-				<a class="post_button" style="bottom: 1px;position:relative; display:inline-block; float:right;" onClick="deletePostToggle({$ud_toggle_status}, {$post['thread_id']}, {$post['id']}, '{$post['author_user_id']}', '{$function_date($date,'n/j g:i A')}', '{$this->core->getCsrfToken()}' )" title="{$ud_button_title}"><i class="fa {$ud_button_icon}" aria-hidden="true"></i></a>
-HTML;
-			}
-		}
-		if($this->core->getUser()->getGroup() <= 3 || $post['author_user_id'] === $current_user) {
-			if(!($this->core->getQueries()->isThreadLocked($thread_id) != 1 || $this->core->getUser()->accessFullGrading() )){
 
-			} else {
+				$post_button["delete"] = Array(
+				    "ud_toggle_status" => $ud_toggle_status,
+                    "csrf_token" => $this->core->getCsrfToken(),
+                    "ud_button_title" => $ud_button_title,
+                    "ud_button_icon" => $ud_button_icon
+                );
+
 				$shouldEditThread = null;
+
 				if($first) {
 					$shouldEditThread = "true";
 					$edit_button_title = "Edit thread and post";
 				} else {
 					$shouldEditThread = "false";
 					$edit_button_title = "Edit post";
 				}
-				$return .= <<<HTML
-					<a class="post_button" style="position:relative; display:inline-block; color:black; float:right;" onClick="editPost({$post['id']}, {$post['thread_id']}, {$shouldEditThread}, '{$this->core->getCsrfToken()}')" title="{$edit_button_title}"><i class="fas fa-edit" aria-hidden="true"></i></a>
-HTML;
+
+				$post_button["edit"] = Array(
+				    "shouldEditThread" => $shouldEditThread,
+                    "edit_button_title" => $edit_button_title,
+                    "csrf_token" => $this->core->getCsrfToken()
+                );
 			}
         }
 
-        $return .= <<<HTML
-		<h7 style="position:relative; right:5px;">
-			<strong id="post_user_id">{$visible_username}</strong>
-			{$function_date($date, "n/j g:i A")}
-HTML;
-        if (!is_null($edit_date)) {
-            $return .= <<<HTML
-			(<i>Last edit at {$edit_date}</i>)
-HTML;
-        }
-        $return .= <<<HTML
-		</h7>
-		</span>
-HTML;
+		$post_attachment = Array("exist" => false);
 
         if ($post["has_attachment"]) {
+            $post_attachment["exist"] = true;
+
             $post_dir = FileUtils::joinPaths($thread_dir, $post["id"]);
             $files = FileUtils::getAllFiles($post_dir);
+
+            $post_attachment["files"] = Array();
+
             foreach ($files as $file) {
                 $path = rawurlencode($file['path']);
                 $name = rawurlencode($file['name']);
                 $name_display = htmlentities(rawurldecode($file['name']), ENT_QUOTES | ENT_HTML5, 'UTF-8');
```

Comment:
get rid of the `htmlentities` from the PHP code and let Twig handle this for us.